query_server/query/src/metadata.rs

```diff
@@ -166,6 +176,183 @@ impl MetaData for LocalCatalogMeta {
+
+        let schema = Arc::new(Schema::new(vec![Field::new(
+            "Database",
+            DataType::Utf8,
+            false,
+        )]));
+
+        let batch = RecordBatch::try_new(schema, vec![Arc::new(StringArray::from(dbs))]).unwrap();
+
+        let batches = vec![Arc::new(batch)];
+
+        Ok(Output::StreamData(stream_from_batches(batches)))
+    }
+
+    fn show_table(&self, name: &str) -> Result<Output> {
+        let mut database_name = name;
+        if database_name.is_empty() {
+            database_name = self.database_name.as_str()
+        }
+
+        match self.catalog.schema(database_name) {
+            None => Err(MetadataError::DatabaseNotExists {
+                database_name: database_name.to_string(),
+            }),
+            Some(_db_cfgs) => {
+                let schema = Arc::new(Schema::new(vec![Field::new(
+                    "Table",
+                    DataType::Utf8,
+                    false,
+                )]));
+
+                let tables = self.engine.list_tables(database_name).unwrap();
+                let batch = RecordBatch::try_new(schema, vec![Arc::new(StringArray::from(tables))])
+                    .unwrap();
+
+                let batches = vec![Arc::new(batch)];
+
+                Ok(Output::StreamData(stream_from_batches(batches)))
+            }
+        }
+    }
+
+    fn describe_database(&self, name: &str) -> Result<Output> {
+        match self.engine.get_db_schema(name) {
+            None => Err(MetadataError::DatabaseNotExists {
+                database_name: name.to_string(),
+            }),
+            Some(db_cfg) => {
+                let schema = Arc::new(Schema::new(vec![
+                    Field::new("TTL", DataType::Utf8, false),
+                    Field::new("SHARD", DataType::Utf8, false),
+                    Field::new("VNODE_DURATION", DataType::Utf8, false),
+                    Field::new("REPLICA", DataType::Utf8, false),
+                    Field::new("PRECISION", DataType::Utf8, false),
+                ]));
+
+                let ttl = db_cfg.config.ttl.to_string();
+                let shard = db_cfg.config.shard_num.to_string();
+                let vnode_duration = db_cfg.config.vnode_duration.to_string();
+                let replica = db_cfg.config.replica.to_string();
+                let precision = db_cfg.config.precision.to_string();
+
+                let batch = RecordBatch::try_new(
+                    schema,
+                    vec![
+                        Arc::new(StringArray::from(vec![ttl.as_str()])),
+                        Arc::new(StringArray::from(vec![shard.as_str()])),
+                        Arc::new(StringArray::from(vec![vnode_duration.as_str()])),
+                        Arc::new(StringArray::from(vec![replica.as_str()])),
+                        Arc::new(StringArray::from(vec![precision.as_str()])),
+                    ],
+                )
+                .unwrap();
```

Comment:
Handle error

Reply:
normally, it will not happen error here

Reply:
Use `unwrap_unchecked` with unsafe if you can ensure that no errors will occur at any time and that there will be no problems due to changes in other places.

Reply:
suggest not to use 'unwrap' here, especially call the interface of datafusion, no guarantee of "no error situation"